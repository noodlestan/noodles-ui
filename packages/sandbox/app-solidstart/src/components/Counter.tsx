'use client';
import { JSX, createSignal } from 'solid-js';

import styles from './Counter.module.css';

export default function Counter(): JSX.Element {
    const [count, setCount] = createSignal(0);
    return (
        <button class={styles.Counter} onClick={() => setCount(count() + 1)}>
            Clicks: {count()}
        </button>
    );
}
