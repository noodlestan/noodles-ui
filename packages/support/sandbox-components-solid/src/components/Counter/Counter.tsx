'use client';
import { Component, createSignal } from 'solid-js';

import styles from './Counter.module.scss';

export const Counter: Component = () => {
    const [count, setCount] = createSignal(0);
    return (
        <button class={styles.Counter} onClick={() => setCount(count() + 1)}>
            Clicks: {count()}
        </button>
    );
};
