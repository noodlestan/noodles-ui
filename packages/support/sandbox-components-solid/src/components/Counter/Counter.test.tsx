import { render } from '@solidjs/testing-library';
import { describe, expect, test } from 'vitest';

import { Counter } from './Counter';

describe('<Counter />', () => {
    test('renders', () => {
        const { container, unmount } = render(() => <Counter />);
        expect(container.innerHTML).toMatchSnapshot();
        unmount();
    });
});
