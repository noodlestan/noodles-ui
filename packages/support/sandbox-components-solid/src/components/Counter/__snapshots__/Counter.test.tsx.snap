// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Counter /> > renders 1`] = `"<button class="_Counter_c43b9e">Clicks: 0</button>"`;
